### PR TITLE
[#27] [Backend] As a user, I can see Survey Detail

### LINKED ISSUE #27: [KMM] [Backend] As a user, I can see Survey Detail
## Why

The user can start the survey process by tapping the `Start Survey` button to return to go to the survey selection screen.

## Acceptance Criteria?

Call endpoint:
`/api/v1/surveys/{id}`
```
Header
`Authorization` : {{user_token_type}} {{user_access_token}}
```

## Resources

[Postman - Nimble Survey API](https://documenter.getpostman.com/view/11835486/T1LJjTqj#142b8597-e176-4e74-95a5-cb48ed5115f0)

### PR DESCRIPTION
**Pull request by blyscuit**
- close #27 

## What happened

Add backend to call survey detail.
 
## Insight

- Add `SurveyDetailRepo`.

### CODE REVIEW

File: shared/src/commonMain/kotlin/co/nimblehq/blisskmmic/data/model/SurveyDetailApiModel.kt [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+import co.nimblehq.blisskmmic.domain.model.SurveyDetail
+import co.nimblehq.blisskmmic.domain.model.Token
+import kotlinx.serialization.SerialName
+import kotlinx.serialization.Serializable
+
+@Serializable
+data class SurveyDetailApiModel(
+    val title: String,
+    val description: String,
+    @SerialName("is_active") val isActive: Boolean,
```
Comment (doannimble) [inline shared/src/commonMain/kotlin/co/nimblehq/blisskmmic/data/model/SurveyDetailApiModel.kt:12]: @blyscuit Can we have the linebreak here?
```suggestion
    @SerialName("is_active") 
    val isActive: Boolean,
```

File: shared/src/commonMain/kotlin/co/nimblehq/blisskmmic/data/repository/SurveyRepositoryImpl.kt [hunk 2]
```diff
@@ -20,4 +22,9 @@ class SurveyRepositoryImpl(private val networkDataSource: NetworkDataSource): Su
                 )
             }
     }
+
+    override fun surveyDetail(id: String): Flow<SurveyDetail> {
```
Comment (doannimble) [inline shared/src/commonMain/kotlin/co/nimblehq/blisskmmic/data/repository/SurveyRepositoryImpl.kt]: @blyscuit just a suggestion, should we put the prefix with `action`? such as `getSurveyDetail`, `featchSuveryDetail`,etc...? Using `surveyDetail` make me confused tbo 🙏

**Review by doannimble — COMMENTED** [review]
Just a few tiny concerns, rest lgtm

File: shared/src/commonMain/kotlin/co/nimblehq/blisskmmic/data/repository/SurveyRepositoryImpl.kt (hunk 2 above)
Comment (blyscuit) [reply shared/src/commonMain/kotlin/co/nimblehq/blisskmmic/data/repository/SurveyRepositoryImpl.kt]: Good catch, I will modify this repository. 🙏

File: shared/src/commonMain/kotlin/co/nimblehq/blisskmmic/data/model/SurveyDetailApiModel.kt (hunk 1 above)
Comment (blyscuit) [reply shared/src/commonMain/kotlin/co/nimblehq/blisskmmic/data/model/SurveyDetailApiModel.kt:12]: Please refer to this [discussion](https://github.com/blyscuit/kmm-swiftUI-IC/pull/106#discussion_r1065309629). We should derive some format for the team, but for this project we will keep this format. 🙏

**Review by doannimble — APPROVED** [review]
lgtm now 💯

File: shared/src/commonTest/kotlin/co/nimblehq/blisskmmic/data/datasource/NetworkDataSourceTest.kt [hunk 3]
```diff
@@ -129,4 +131,22 @@ class NetworkDataSourceTest {
                 awaitComplete()
             }
     }
+
+    // Survey Detail
+
+    @Test
+    fun `When calling survey detail with success response- it returns correct object`() = runTest {
```
Comment (minhnimble) [inline shared/src/commonTest/kotlin/co/nimblehq/blisskmmic/data/datasource/NetworkDataSourceTest.kt]:
```suggestion
    fun `When calling survey detail with success response - it returns correct object`() = runTest {
```

File: shared/src/commonMain/kotlin/co/nimblehq/blisskmmic/data/model/SurveyDetailApiModel.kt [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+    @SerialName("cover_image_url") val coverImageUrl: String,
+    @SerialName("survey_type") val surveyType: String,
+    val questions: List<SurveyIncluded>
+    ) {
```
Comment (minhnimble) [inline shared/src/commonMain/kotlin/co/nimblehq/blisskmmic/data/model/SurveyDetailApiModel.kt]: I think there is a indentation issue here 🤔
```suggestion
data class SurveyDetailApiModel(
    val title: String,
    val description: String,
    @SerialName("is_active") val isActive: Boolean,
    @SerialName("cover_image_url") val coverImageUrl: String,
    @SerialName("survey_type") val surveyType: String,
    val questions: List<SurveyIncluded>
) {
```

File: shared/src/commonTest/kotlin/co/nimblehq/blisskmmic/data/repository/SurveyRepositoryTest.kt [hunk 5]
```diff
@@ -55,9 +58,24 @@ class SurveyRepositoryTest: TestsWithMocks() {
...
         surveyRepository
-            .survey(1)
+            .getSurvey(1)
             .test {
                 awaitError().message shouldBe "Fail"
             }
     }
+
+    @Test
+    fun `When calling surveyDetail with success response- it returns correct object`() = runTest {
```
Comment (minhnimble) [inline shared/src/commonTest/kotlin/co/nimblehq/blisskmmic/data/repository/SurveyRepositoryTest.kt]:
```suggestion
    fun `When calling surveyDetail with success response - it returns correct object`() = runTest {
```

**Review by minhnimble — COMMENTED** [review]
Just some small refactor points left, rest LGTM 🚀

**Review by minhnimble — APPROVED** [review]
LGTM 🚀

**Pull request merged**